policies/templates/gcp_restricted_firewall_rules_v1.yaml

```diff
@@ -0,0 +1,628 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+#
+# This template is for policies restricting which firewall rules
+# should exist in a specific location in GCP.
+# All firewall rules matching the passed parameters will be flagged as violations.
+# All parameters are optional.
+#
+# A firewall rule will trigger a violation ONLY IF it matches ALL the passed parameters.
+
+apiVersion: templates.gatekeeper.sh/v1alpha1
+kind: ConstraintTemplate
+metadata:
+  name: gcp-restricted-firewall-rules-v1
+spec:
+  crd:
+    spec:
+      names:
+        kind: GCPRestrictedFirewallRulesConstraintV1
+        plural: gcprestrictedfirewallrulesconstraintsv1
+      validation:
+        openAPIV3Schema:
+          properties:
+            rules:
+              description: "List of rules to match in order to raise violations. Rule are object describing rules (see below)"
+              type: array
+              items: objects
+              additionalProperties:
+                direction:
+                  description: "Direction of the rules to restrict."
+                  type: string
+                  enum: ["INGRESS", "EGRESS", "any"]
+                  default: "any"
+                rule_type:
+                  description: "Type of the rules to restrict."
+                  type: string
+                  enum: ["allowed", "denied", "any"]
+                  default: "any"
+                port:
```

Comment:
Should this be ports?

Reply:
I think so, I matched the name in the original CAI exports but they got updated. Will fix.

Reply:
Actually, currently the rule only support one port:protocol pair as a parameter. 

I could add support for more than one but now that we added the "rules: " top-level object it's less needed, I think.

Reply:
Got it, let's just make sure the schema is correct then.

Reply:
I updated the schema to fix this (and protocol) and update the properties for rule.